pkg/util/json.go

```diff
@@ -0,0 +1,148 @@
+// where the error appeared in the json file which was validated.
+// It contains the fileName, line number, character number, and line
+// content as additional information. Furthermore, it contains the original
+// error (the cause) which happened during the json unmarshalling.
+type JsonValidationError struct {
+
+	// FileName is the file name (full qualified) where the error happened
+	// This field is always filled.
+	FileName string
+	// LineNumber contains the line number (starting by one) where the error happened
+	// If we don't have the information, this is -1.
+	LineNumber int
+	// CharacterNumberInLine contains the character number (starting by one) where
+	// the error happened. If we don't have the information, this is -1.
+	CharacterNumberInLine int
+	// LineContent contains the full line content of where the error happened
+	// If we don't have the information, this is an empty string.
+	LineContent string
+	// PreviousLineContent contains the full line content of the line before LineContent
+	// If we don't have the information, this is an empty string.
+	PreviousLineContent string
+	// Cause is the original error which happened during the json unmarshalling.
+	Cause error
+}
+
+func (e JsonValidationError) Error() string {
+	return fmt.Sprintf("file %s is not a valid json: Error: %s", e.FileName, e.Cause.Error())
+}
+
+// ContainsLineInformation indicates whether additional line information is present in
+// the error.
+func (e *JsonValidationError) ContainsLineInformation() bool {
+	return e.LineNumber > 0 && e.CharacterNumberInLine > 0 && e.LineContent != ""
+}
+
+const errorTemplate = `File did not contain valid json:
+ --> %s:%d:%d
+ %s | %s
+ %d | %s
+ %s | %s^^^
+ %s - Cause: %s
+`
+
+func (e *JsonValidationError) PrettyPrintError() {
+
+	if e.ContainsLineInformation() {
+
+		lengthOfLineNum := len(strconv.Itoa(e.LineNumber))
+		whiteSpace := strings.Repeat(" ", lengthOfLineNum)
+		whiteSpaceOffset := strings.Repeat(" ", e.CharacterNumberInLine-1)
+		lineContent := strings.Replace(e.LineContent, "\t", " ", -1)
+		previousLineContent := strings.Replace(e.PreviousLineContent, "\t", " ", -1)
+
+		Log.Error(errorTemplate, e.FileName, e.LineNumber, e.CharacterNumberInLine,
+			whiteSpace, previousLineContent,
+			e.LineNumber, lineContent,
+			whiteSpace, whiteSpaceOffset,
+			whiteSpace, e.Cause.Error())
+	}
+}
+
+// ValidateJson validates whether the json file is correct, by using the internal validation done
+//when unmarshalling to a an object. As none of our jsons can actually be unmarshalled
+//to a string, we catch that error, but report any other error as fatal.
+func ValidateJson(jsonString string, filename string) error {
+
+	var j string
+	err := json.Unmarshal([]byte(jsonString), &j)
+
+	if jsonError, ok := err.(*json.SyntaxError); ok {
+		return mapError(jsonString, filename, int(jsonError.Offset), err)
+	}
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// TODO actually check against the model
```

Comment:
This `TODO` is the only issue I have with this PR ;) 
Is this to do as part of you still working on it, or a future enhancement? 
If the first, I'll wait to accept the PR, if the latter could you create and issue and reference it in this TODO please ?

Reply:
:smile: No, the TODO is for a future enhancement. We don't have any model information currently.

Reply:
Ok, could you create an issue for it and reference the id in the TODO comment, then please?

Todo's in code are way to easy to forget about :D

Reply:
Just created #104.

Reply:
... and added it to the comment.

Reply:
... and I lost the approval because of that. :sob: